src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttKafkaMapper.java

```diff
@@ -69,20 +65,25 @@ public String map(String mqttTopic) {
 
             if (matcher.matches()) {
                 HashMap<String, String> placeholders = new HashMap<>();
+                Pattern placeholderPattern = Pattern.compile(MQTT_TOPIC_PLACEHOLDER_REGEX);
```

Comment:
while compiling the pattern every time you run the `map` method and there is a match?
Just compile it one time in the constructor and you are happy :-)
In the end it's like to be part of the `buildRegex` you already call there.

Reply:
Marked as fixed.